modules/on_gke/main.tf

```diff
@@ -14,61 +14,149 @@
  * limitations under the License.
  */
 
-//*****************************************
-//  Create Keys for Forseti server and
-//  client service accounts.
-//*****************************************
+resource "random_id" "random_hash_suffix" {
+  byte_length = 4
+}
+
+resource "random_integer" "random_minute" {
+  min = 0
+  max = 59
+}
+
+resource "null_resource" "org_id_and_folder_id_are_both_empty" {
+  count = length(var.composite_root_resources) == 0 && var.org_id == "" && var.folder_id == "" ? 1 : 0
+
+  provisioner "local-exec" {
+    command     = "echo 'composite_root_resources=${var.composite_root_resources} org_id=${var.org_id} folder_id=${var.org_id}' >&2; false"
+    interpreter = ["bash", "-c"]
+  }
+}
 
-resource "google_service_account_key" "server_key" {
-  service_account_id = var.forseti_server_service_account
+resource "null_resource" "email_without_sendgrid_api_key" {
+  count = var.inventory_email_summary_enabled == "true" && var.sendgrid_api_key == "" ? 1 : 0
+
+  provisioner "local-exec" {
+    command     = "echo 'inventory_email_summary_enabled=${var.inventory_email_summary_enabled} sendgrid_api_key=${var.sendgrid_api_key}' >&2; false"
+    interpreter = ["bash", "-c"]
+  }
 }
 
-resource "google_service_account_key" "client_key" {
-  service_account_id = var.forseti_client_service_account
+#--------#
+# Locals #
+#--------#
+locals {
+  kubernetes_namespace = "${var.k8s_forseti_namespace}-${local.random_hash}"
+
+  random_hash     = var.resource_name_suffix == null ? random_id.random_hash_suffix.hex : var.resource_name_suffix
+  network_project = var.network_project != "" ? var.network_project : var.project_id
+
+  services_list = [
+    "admin.googleapis.com",
+    "appengine.googleapis.com",
+    "bigquery-json.googleapis.com",
+    "cloudbilling.googleapis.com",
+    "cloudresourcemanager.googleapis.com",
+    "sql-component.googleapis.com",
+    "sqladmin.googleapis.com",
+    "compute.googleapis.com",
+    "iam.googleapis.com",
+    "cloudtrace.googleapis.com",
+    "container.googleapis.com",
+    "containerregistry.googleapis.com",
+    "servicemanagement.googleapis.com",
+    "logging.googleapis.com",
+    "cloudasset.googleapis.com",
+    "storage-api.googleapis.com",
+    "groupssettings.googleapis.com",
+  ]
+  workload_identity               = "${var.project_id}.svc.id.goog"
+  workload_identity_server_suffix = "[${local.kubernetes_namespace}/forseti-server]"
+  workload_identity_client_suffix = "[${local.kubernetes_namespace}/forseti-orchestrator]"
+
+  forseti_run_frequency = var.forseti_run_frequency == null ? "${random_integer.random_minute.result} */2 * * *" : var.forseti_run_frequency
 }
 
 //*****************************************
 //  Allow GKE Service Account to read GCS
 //*****************************************
 
 resource "google_project_iam_member" "cluster_service_account-storage_reader" {
+  count   = var.gke_service_account != "default" ? 1 : 0
```

Comment:
This var.gke_service_account variable does not seem to have a "default" value that I can find. Is this the intended behavior?

Reply:
Yes.  Interestingly when you create a GKE cluster with the "default" service account, either with Terraform, gcloud or the GCP console, the service account name returned by the GKE data source is "default".